### DISCUSSION TITLE
Create new app, fails to serve locally with UNABLE_TO_GET_ISSUER_CERT_LOCALLY

### DISCUSSION THREAD

**Discussion by qd-danh** [observablehq/framework#1874] [Q&A] [answered]
Windows 10, command prompt

Create a new framework app with `npx "@observablehq/framework@latest" create`

Change directory into newly created app and `npm run dev`

See only `fetch failed` in launched browser. 

See the following output in the Windows command prompt console (similar block repeats 4 times each with the same error cause and code)

```
GET /
npm:@observablehq/plot@0.6.16/+esm → src/.observablehq/cache/_npm/@observablehq/plot@0.6.16/_esm.js
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11576:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///D:/Projects/Learning/ObsFramework/itj-holiday-2024/node_modules/@observablehq/framework/dist/npm.js:75:22 {
  cause: Error: unable to get local issuer certificate
      at TLSSocket.onConnectSecure (node:_tls_wrap:1550:34)
      at TLSSocket.emit (node:events:514:28)
      at TLSSocket._finishInit (node:_tls_wrap:967:8)
      at ssl.onhandshakedone (node:_tls_wrap:743:12) {
    code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY'
  }
}
```

Something wrong with latest app creation logic? 

I have another framework app that was created a couple weeks ago, running locally just fine with local web server hosting. 

Tried creating a new framework app with older version (same one that is working for me in the other app) and get the same problem. `npx "@observablehq/framework@1.11.0" create`

**Comment by Fil** [comment]
This post seems relevant:
https://superuser.com/questions/1789411/is-unable-to-get-issuer-cert-locally-a-firewall-company-policy-issue

let us know if any of the tips provided in the top answer works for you?

**Reply by qd-danh** [reply]
I should have mentioned that I tried things like those suggested in the article and summary. 

As I mentioned, existing framework project created a couple weeks ago works (`npm run dev`) no problem. I'm assuming that's because it fetched all necessary npm packages and now no longer has to.

`npm config set strict-ssl false` => no change
`npm config set registry http://registry.npmjs.org/` => no change
`export NODE_TLS_REJECT_UNAUTHORIZED=0` => different error, unable to fetch ... so maybe we're on to something... 

`Error: unable to fetch: https://cdn.jsdelivr.net/npm/@observablehq/plot@0.6.16/+esm`
I try to fetch that specific file manually in browser and get an outbound corporate firewall blocked page.

Will investigate with our IT secops peeps

**Comment by qd-danh** [comment, marked as answer]
OK - solved this one. 
Turns out IT at my company had blocked `cdn.jsdelivr.net` due to what apparently was **jsdelivr.net being used in a malware attack on December 5th **  

Sounds like it was since "cleaned up" so they are able to unblock that site. 

`npm run dev` and the web serving is working fine for me again.

Strange manifestation of the problem with the rabbit whole errors about `UNABLE_TO_GET_ISSUER_CERT_LOCALLY` but I suppose that was a problem downloading the cert

**Comment by LyraHarring** [comment]
You can check out the following resource guide for the error solution and follow the steps accordingly:- https://cheapsslweb.com/blog/fixing-unable-to-get-issuer-cert-locally-error/. It emerges when an SSL certificate cannot be verified or confirmed due to the absence or lack of the issuer’s certificate in the local trust store.

I hope it helps!
